### PR TITLE
Feat(frontend): Created checkout page

### PR DESCRIPTION
**Pull request by NightFury742**
## Overview

1. This PR fixes or fixes part of #41.
2. This PR does the following: 
- Created the checkout page (responsive only for large screen devices).
- Added route to it when user clicks  the pay now button on the bus details drawer.
- Added a toast which displays an error when user clicks on the button without adding a passenger.
- Made passenger details in the drawer dynamic and created a separate component for it (`BusDetailsCard`).
- Made fare breakdown details in the drawer dynamic and created a separate component for it (`FareBreakdownCard`).
- Changed the `types.d.ts` file to account for different state's types.
- Used **Zustand** to create a orderStore to manage states across the different components effectively.

 

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [ ] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct

https://github.com/bsoc-bitbyte/busify/assets/119070798/6c26a1ec-2e53-4b56-aedb-463d783dce33

### CODE REVIEW

**Comment by NightFury742** [comment]
@parthksingh1 will be working on the ticket breakdown and ticket total component in the checkout page.

**Comment by NightFury742** [comment]
@prajjwalkapoor @Bhavik-ag  Please review my pr.
 This is how the website looks,

https://github.com/bsoc-bitbyte/busify/assets/119070798/6c49e212-2c93-42df-be34-582cb22b70c3

File: frontend/src/pages/Checkout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                  color={theme.palette.secondary.main}
+                >
+                  Processing Fee
+                </Typography>
+                <Box display="flex">
+                  <Typography fontSize="1.1rem" fontWeight="600">
+                    &#x20B9;
+                  </Typography>
+                  <Typography fontSize="1.1rem" fontWeight="600">
+                    {Math.ceil(ticketQuantity * 20 * 0.02)}
```
Comment (prajjwalkapoor) [inline frontend/src/pages/Checkout/index.tsx:161]: Don't hardcode the price. It comes with a schedule from the backend. Use that.

File: frontend/src/pages/Checkout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  margin: 2rem 0;
+  padding: 1rem;
+  min-height: 20vh;
+`;
+
+const BusDetails = () => {
+  const passengerDetail = useOrderStore(state => state.passengerDetail);
+  const ticketQuantity = useOrderStore(state => state.ticketQuantity);
+  const theme = useTheme();
+  return (
```
Comment (prajjwalkapoor) [inline frontend/src/pages/Checkout/index.tsx:37]: Make the checkout page mobile responsive.

**Review by prajjwalkapoor — CHANGES_REQUESTED** [review]
Nice work @NightFury742! 

I have suggested some changes, just do that.

**Comment by NightFury742** [comment]
@prajjwalkapoor I have made the requested changes, please review them.

**Review by prajjwalkapoor — APPROVED** [review]
LGTM.

**Pull request merged**